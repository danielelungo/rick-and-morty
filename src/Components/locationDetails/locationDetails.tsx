import React from "react";
import { Character, CharacterLocation } from "../../Interfaces/types";
import CharDetail from "../charDetails/charDetails";

interface LocationDetailsProps {
  mainData?: CharacterLocation;
  error: string;
  response: Character;
}

const LocationDetails: React.FC<LocationDetailsProps> = ({
  mainData,
  error,
  response,
}) => {
  return (
    <>
      <CharDetail
        dataTestid="Location"
        data={response?.name}
        title="Location: "
      />

      {error ? (
        <span>{error}</span>
      ) : (
        mainData && (
          <div>
            <CharDetail
              data={mainData?.residents?.length}
              title="Amount of residents: "
              dataTestid="amount of residents"
              list
            />

            <CharDetail
              data={mainData?.dimension}
              title="Dimension: "
              dataTestid="dimension"
              list
            />
            <CharDetail
              dataTestid="type"
              data={mainData?.type}
              title="Type: "
              list
            />
          </div>
        )
      )}
    </>
  );
};

export default LocationDetails;
