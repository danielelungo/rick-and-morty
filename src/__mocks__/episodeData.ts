export const episodeData = [
  { date: "December 2, 2013", id: 1, name: "Pilot", episode: "S01E01" },
  { date: "December 9, 2013", id: 2, name: "Lawnmower Dog", episode: "S03E03" },
];
