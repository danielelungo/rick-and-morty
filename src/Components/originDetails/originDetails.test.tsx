import React from "react";
import renderer from "react-test-renderer";
import { episodeDetails } from "../../__mocks__/episodeDetails";
import { locationDetails } from "../../__mocks__/locationDetails";
import OriginDetails from "./origindetails";

describe("OriginDetails component", () => {
  test("renders OriginDetails Component", () => {
    const tree = renderer
      .create(
        <OriginDetails
          mainData={locationDetails}
          error={""}
          response={episodeDetails}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
