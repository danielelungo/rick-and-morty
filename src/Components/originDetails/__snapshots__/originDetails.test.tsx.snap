// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OriginDetails component renders OriginDetails Component 1`] = `
Array [
  <div
    className="sc-bczRLJ gVHrnU"
  >
    <h4
      className="sc-dkzDqf"
    >
      Origin: 
       
      unknown
    </h4>
  </div>,
  <div>
    <div
      className="sc-bczRLJ gVHrnU"
    >
      <li
        className="sc-gsnTZi fMdjaC"
      >
        Amount of residents: 
         
        100
      </li>
    </div>
    <div
      className="sc-bczRLJ gVHrnU"
    >
      <li
        className="sc-gsnTZi fMdjaC"
      >
        Dimension: 
         
        unknown
      </li>
    </div>
    <div
      className="sc-bczRLJ gVHrnU"
    >
      <li
        className="sc-gsnTZi fMdjaC"
      >
        Type: 
         
        Space station
      </li>
    </div>
  </div>,
]
`;
